backend/kernelCI_app/views/hardwareDetailsView.py

```diff
@@ -190,8 +203,24 @@ def pass_in_filters(self, data, filters):
 
         return True
 
+    def should_jump_test(
+        self,
+        table_test: Literal["boot", "test"],
+        record: Dict,
+        filters_map: Dict[str, List[Dict]],
+        processed_test: Set[str],
+    ) -> bool:
+        has_test_been_processed = record["id"] in processed_test
+        if has_test_been_processed:
+            return True
+
+        test_filter_is_empty = len(filters_map[table_test]) == 0
+        test_pass_in_filters = self.pass_in_filters(record, filters_map[table_test])
+        return not test_filter_is_empty and not test_pass_in_filters
+
     def sanitize_records(self, records, selected_trees, filters):
-        processed_builds = set()
+        processed_builds: Set[str] = set()
+        processed_test: Set[str] = set()
```

Comment:
There is no need need to mark the processed tests since each record is a unique test

Reply:
nit: we can rename the records variable to make this fact more explicit

Reply:
we are using build_id as the key for the record check, and it is only validating in the build handling logic
this add the test and validates inside the test handling logic

Reply:
This is making no sense, and the prove is the fact that you are never populating `processe_test`.

Reply:
oh got it, this is in an inner join from tests